src/InstrumentationEngine/ProfilerManager.cpp

```diff
@@ -506,7 +507,9 @@ HRESULT MicrosoftInstrumentationEngine::CProfilerManager::AddInstrumentationMeth
             // InstrumentationMethod. Since we are using global lock - InstrumentationMethod can only set flags associated
             // with it in this thread.
             CInitializeInstrumentationMethodHolder initHolder(this);
-            hr = pInstrumentationMethod->Initialize(this, m_bValidateCodeSignature);
+            CComPtr<CProfilerManagerWrapper> pProfilerManagerWrapper;
+            pProfilerManagerWrapper.Attach(new (nothrow) CProfilerManagerWrapper(pInstrumentationMethod->GetClassId(), this));
```

Comment:
Since you have "nothrow", you should check that pProfilerManagerWrapper is not null: IfNullRet(pProfilerManagerWrapper);

Reply:
And return E_OUTOFMEMORY

---
In reply to: [331293268](https://github.com/microsoft/CLRInstrumentationEngine/pull/146#discussion_r331293268) [](ancestors = 331293268)

Reply:
Thanks!

---
In reply to: [331673239](https://github.com/microsoft/CLRInstrumentationEngine/pull/146#discussion_r331673239) [](ancestors = 331673239,331293268)